src/v/cloud_storage_clients/tests/client_pool_mt_test.cc

```diff
@@ -218,7 +220,8 @@ SEASTAR_THREAD_TEST_CASE(test_client_pool_acquire_after_leasing_all) {
         cloud_storage_clients::client_pool_overdraft_policy::borrow_if_empty)
       .get();
 
-    pool.invoke_on_all(&cloud_storage_clients::client_pool::start).get();
+    pool.invoke_on_all(&cloud_storage_clients::client_pool::start, std::nullopt)
+      .get();
```

Comment:
nitpick: any particular reason to change all of these call sites when we have a default arg already?